cme/modules/nanodump.py

```diff
@@ -51,22 +65,36 @@ def options(self, context, module_options):
     def on_admin_login(self, context, connection):
         if self.useembeded == True:
             with open(self.nano_path + self.nano, 'wb') as nano:
-                nano.write(self.nano_embeded)
+                # Figure out if we're dealing with 32-bit or 64-bit Windows
+                command = 'wmic os get osarchitecture | find /v "OSArchitecture"'
```

Comment:
since we already have the os arch of the target in https://github.com/byt3bl33d3r/CrackMapExec/blob/master/cme/protocols/smb.py#L126, we could simplify by doing: `connection.os_arch`

Reply:
Excusez-moi, I seem to have overlooked the context I was in. Changed it 🙂